### PR TITLE
Chat Logging & Other

### PR DESCRIPTION
**Pull request by radongc**
- Add player chat logging based on vmangos log structure, stored in `etc/logs/chat.log`
- Added `summon <player>` and `port <player>` console commands
- Port will teleport you to a player if one is found, otherwise it will try to teleport you to a .tel location

### CODE REVIEW

File: utils/PlayerLogManager.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+                log_type = 'Emote'
+            elif chat_type == ChatMsgs.CHAT_MSG_PARTY:
+                log_type = 'Party'
+            elif chat_type == ChatMsgs.CHAT_MSG_GUILD:
+                log_type = 'Guild'
+            elif chat_type == ChatMsgs.CHAT_MSG_OFFICER:
+                log_type = 'Officer'
+
+            PlayerLogManager._write_to_log(f'{date} [CHAT] (GUID {player_mgr.guid}, NAME {player_mgr.get_name()}, MAP {player_mgr.map_}, POS {PlayerLogManager._get_location_string(player_mgr)}): [{log_type}] {player_mgr.get_name()}:{player_mgr.guid} : {msg}')
+            return
```
Comment (GrenderG) [inline utils/PlayerLogManager.py:31]: There are some redundant `return`s, no?

**Comment by GrenderG** [comment]
I feel like this should be handled in another thread just for log writing. Maybe a generic `LogManager` with a queue mechanism. File I/O is very slow.

**Comment by radongc** [comment]
> I feel like this should be handled in another thread just for log writing. Maybe a generic `LogManager` with a queue mechanism. File I/O is very slow.

True, I'll refactor and see about making these changes.

**Comment by GrenderG** [comment]
> > I feel like this should be handled in another thread just for log writing. Maybe a generic `LogManager` with a queue mechanism. File I/O is very slow.
> 
> True, I'll refactor and see about making these changes.

Great! Let us know if you have any doubt, you can also check other parts of the core where a queue is implemented.

**Comment by radongc** [comment]
> > > I feel like this should be handled in another thread just for log writing. Maybe a generic `LogManager` with a queue mechanism. File I/O is very slow.
> > 
> > 
> > True, I'll refactor and see about making these changes.
> 
> Great! Let us know if you have any doubt, you can also check other parts of the core where a queue is implemented.

Ready for review now 👍

**Comment by GrenderG** [comment]
@radongc for Channels, ignore for logging the ones that return `True` with `is_addon()`.

**Comment by radongc** [comment]
> @radongc for Channels, ignore for logging the ones that return `True` with `is_addon()`.

In `ChatManager.py`, I made sure that logging is called only if `is_addon()` is false. Should I add that condition in `LogManager.py` as well?

**Comment by radongc** [comment]
> @radongc for Channels, ignore for logging the ones that return `True` with `is_addon()`.

Added a condition for this in `LogManager`. 👍

**Pull request merged**